apps/styleguide/src/forms/InputOntology.vue

```diff
@@ -359,32 +364,51 @@ export default {
       else if (value) {
         let term = this.terms[value.name];
         if (term) {
-          term.selected = "complete";
+          term.selected = 'complete';
           this.getParents(term).forEach((parent) => {
-            parent.selected = "partial";
+            parent.selected = 'partial';
           });
         }
       }
       this.key++;
-    },
+    }
   },
   watch: {
     search() {
-      //first show/hide depending on filter
-      Object.values(this.terms).forEach(
-        (t) => (t.visible = this.search == '' || !this.search)
-      );
-      if (this.search && this.search.length > 0) {
-        let searchTerms = this.search.split(' ').map((s) => s.toLowerCase());
+      this.searchResultCount = 0;
+      if (this.search) {
+        //first hide all
+        Object.values(this.terms).forEach((t) => (t.visible = false));
+        //split and sanitize search terms
+        let searchTerms = this.search
+          .trim()
+          .split(' ')
+          .filter((s) => s.trim().length > 0)
+          .map((s) => s.toLowerCase());
+        //check every term if it matches all search terms
         Object.values(this.terms).forEach((term) => {
           if (searchTerms.every((s) => term.name.toLowerCase().includes(s))) {
-            //items are visible when matching search, or when a child matches search
             term.visible = true;
-            this.getParents(term).forEach((parent) => {
-              parent.visible = true;
-            });
+            this.searchResultCount++;
+
+            //also make parents visible
+            if (term.parent) {
```

Comment:
Here was in previous version the slow bit that resulted in page unresponsive. Before we used getParents().forEach and that was very expensive if you do this for almost all nodes if you for example start your search with 'e' resulting in almost everything matching.